### ISSUE TITLE
Adding suffix to package name if exists another with same name

### ISSUE THREAD

**Issue by dportoles** [ckan/ckan#4637]
If you want to create a package with a name that already exists for another dataset, an error is always raised.
This is a common issue if several datasets has the same title (probably from different orgs with same information) because the name is calculated using title.
Sometimes it is desired that name would been modified but no error would be raised. For instance when the datasets are created from an automatic process with no human intervention.

### CKAN Version if known (or site URL)
Current version

### Please describe the expected behaviour
Yo can choose by defining an .ini property if you want to raises an error or regenerating a new modified dataset name (e.g. adding a suffix)
The default value for this new property could be raises always an error for backwards compatibility.

### Please describe the actual behaviour
Always raises an error

### What steps can be taken to reproduce the issue? 
Create two datasets with same name

**Comment by dportoles** [comment]
I will submit soon a PR with a possible solution for reviewing it

**Comment by mutantsan** [comment]
As I can see, currently we can create packages with duplicate names, but we should provide the unique URL. Is it good idea just to add timestamp to the URL automatically?

**Comment by dportoles** [comment]
@mutantsan No, you can't create packages with same name. It always raises an error.
But sometimes (automatic processes) it's useful to modify name to another name without raising an error.

It should be configurable, of course, maintaining backwards compatibility. 

My proposal is adding a serial rather than a timestamp.
e.g.
1st attemp to create dataset name "foo"  => dataset created with name "foo"
2nd attemp to create dataset name "foo"  =>dataset created with name "foo1" 
3rd attemp to create dataset name "foo"  =>dataset created with name "foo2"
and so on

**Comment by mutantsan** [comment]
@dportoles Yes, my bad. I was talking about title. But still I insist that adding timestamp is better idea, just because this approach is simpler and more pythonic.

**Comment by dportoles** [comment]
@mutantsan Probably more pythonic but IMHO url should be easy to read/understand

**Comment by Gauravp-NEC** [comment]
@dportoles @wardi, I did investigation on this issue. According to me we need to add suffix to a package name which already exists (in same organisation or in another organisation) with same package name. Please confirm my understanding. Thanks

**Comment by Gauravp-NEC** [comment]
@wardi , proposal of adding a serial rather than a timestamp.
e.g.
1st attemp to create dataset name "foo" => dataset created with name "foo"
2nd attemp to create dataset name "foo" =>dataset created with name "foo1"
3rd attemp to create dataset name "foo" =>dataset created with name "foo2"
and so on. Is it the right approach to fix this issue? Please let me know your opinion.

**Comment by dportoles** [comment]
Yes, that should be the right approach

**Comment by Gauravp-NEC** [comment]
@dportoles, I have raised the PR #7425  to fix this issue.
Can you confirm that this issue is fixed by #7425 ? Thanks :)

**Comment by dportoles** [comment]
I'm not working in CKAN at this moment, so it's a bit difficult to be sure...
But I think your approach is ok and for me would be enough!

**Comment by Gauravp-NEC** [comment]
@wardi @smotornyuk , I think we can close this issue in favour of issue https://github.com/ckan/ckan/issues/7481.
